src/cli/subcommand/uninstall.rs

```diff
@@ -111,7 +111,48 @@ impl CommandExecute for Uninstall {
         let install_receipt_string = tokio::fs::read_to_string(receipt)
             .await
             .wrap_err("Reading receipt")?;
-        let mut plan: InstallPlan = serde_json::from_str(&install_receipt_string)?;
+
+        let mut plan: InstallPlan = match serde_json::from_str(&install_receipt_string) {
+            Ok(plan) => plan,
+            Err(plan_err) => {
+                #[derive(serde::Deserialize)]
+                struct MinimalPlan {
+                    version: semver::Version,
+                }
+                let minimal_plan: Result<MinimalPlan, _> =
+                    serde_json::from_str(&install_receipt_string);
+                match minimal_plan {
+                    Ok(minimal_plan) => {
+                        return Err(plan_err).wrap_err_with(|| {
+                            let plan_version = minimal_plan.version;
+                            let current_version = current_version().map(|v| v.to_string()).unwrap_or_else(|_| env!("CARGO_PKG_VERSION").to_string());
+                            format!(
+                            "\
+                            Unable to parse plan, this plan was created by `nix-installer` version `{plan_version}`, this is `nix-installer` version `{current_version}`\n\
+                            To uninstall, either run  `/nix/nix-installer uninstall` or `curl --proto '=https' --tlsv1.2 -sSf -L https://install.determinate.systems/nix/tag/{plan_version} | sh -s -- uninstall`\
```

Comment:
This might not work if they were e.g. using a version from a PR or something... But in most cases this should be fine.

Reply:
I think the only time it would fail is if they installed some `*-unreleased` from a PR then tried to remove from a different PR with the same major/minor/patch. I think this is minimal risk.